src/DataStructures/Tensor/Expressions/TensorAsExpression.hpp

```diff
@@ -179,13 +179,32 @@ struct TensorAsExpression<Tensor<X, Symm, IndexList<Indices...>>,
           typename detail::TensorAsExpressionSymm<Symm, IndexList<Indices...>,
                                                   ArgsList<Args...>>::type,
           IndexList<Indices...>, ArgsList<Args...>> {
+  // === Index properties ===
+  /// The type of the data being stored in the result of the expression
   using type = X;
+  /// The list of \ref SpacetimeIndex "TensorIndexType"s of the result of the
+  /// expression
   using symmetry =
       typename detail::TensorAsExpressionSymm<Symm, IndexList<Indices...>,
                                               ArgsList<Args...>>::type;
+  /// The list of \ref SpacetimeIndex "TensorIndexType"s of the result of the
+  /// expression
   using index_list = IndexList<Indices...>;
-  static constexpr auto num_tensor_indices = tmpl::size<index_list>::value;
+  /// The list of generic `TensorIndex`s of the result of the expression
   using args_list = ArgsList<Args...>;
+  /// The number of tensor indices in the result of the expression
+  static constexpr auto num_tensor_indices = tmpl::size<index_list>::value;
+
+  // === Arithmetic tensor operations properties ===
+  /// The number of arithmetic tensor operations done in the subtree for the
+  /// left operand, which is 0 because this is a leaf expression
+  static constexpr size_t num_ops_left_child = 0;
+  /// The number of arithmetic tensor operations done in the subtree for the
+  /// right operand, which is 0 because this is a leaf expression
+  static constexpr size_t num_ops_right_child = 0;
+  /// The total number of arithmetic tensor operations done in this expression's
+  /// whole subtree, which is 0 because this is a leaf expression
+  static constexpr size_t num_ops_subtree = 0;
```

Comment:
Would it be useful to add tests that check the `num_ops_subtree` of some example expressions is what you would expect?

Reply:
Thanks for the suggestions! I've added tests like you've described to `Test_AddSubtract.cpp`, `Test_Contract.cpp`, `Test_Divide.cpp`, `Test_Negate.cpp`, `Test_Product.cpp`, and `Test_SquareRoot.cpp` in functions called `test_tensor_ops_properties()`. `TensorAsExpression` and `NumberAsExpression` are indirectly also tested throw their usage in those tests.

Reply:
Now that I think about it, I think the `get_used_for_size` functions I've added to each derived TE class probably isn't needed anymore for the same reason - not unless there is an expected performance difference between the following two cases:

(a)
```
tnsr::i<DataVector> R(used_for_size);
get<0>(R) = ...;
get<1>(R) = ...;
get<2>(R) = ...;
```

(b)
```
tnsr::i<DataVector> R{};
get<0>(R) = ...;
get<1>(R) = ...;
get<2>(R) = ...;
```

Do either you or @nilsdeppe know if (a) or (b) is generally preferred? If not, it's something I could look into